### PR TITLE
fix(common): temporarily re-export and deprecate `XhrFactory`

### PR DESCRIPTION
**Pull request by petebacondarwin**
Docs show deprecation notice: https://pr41393-f625e20.ngbuilds.io/api/common/http/XhrFactory
Tested against ngcc-validation: https://github.com/angular/ngcc-validation/pull/2820
Deprecation message also shows in IDE:
Screenshot 2021-03-31 at 17 12 25

### CODE REVIEW

**Review by atscott — APPROVED** [review]
reviewed-for: public-api

File: packages/misc/angular-in-memory-web-api/src/in-memory-web-api-module.ts [hunk 1]
```diff
@@ -6,8 +6,7 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {XhrFactory} from '@angular/common';
-import {HttpBackend} from '@angular/common/http';
+import {HttpBackend, XhrFactory} from '@angular/common/http';
```
Comment (petebacondarwin) [inline packages/misc/angular-in-memory-web-api/src/in-memory-web-api-module.ts:9]: I added this to check that the old import still works locally.
We should probably revert this change before merging.

**Review by alan-agius4 — APPROVED** [review]
LGTM, not sure why this didn't work for me when I tried it. Clearly, I was doing something different.

**Review by gkalpak — APPROVED** [review]
:+1:

Shouldn't we also add a note in [aio/content/guide/deprecations.md][1]?

[1]: https://github.com/angular/angular/blob/f69734eb7618a6493fc98684b3967ce892ba7b2a/aio/content/guide/deprecations.md

**Review by AndrewKushnir — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:366151660:BASE:366151675:1617236971007:dffea5b8).

**Review by jelbourn — APPROVED** [review]
LGTM

Reviewed-for: docs-packaging-and-releasing

**Review by jelbourn — APPROVED** [review]
LGTM

Reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
@petebacondarwin FYI presubmits are successful for the changes in this PR.
